### PR TITLE
LF 2508 Update Sensor Backend/ Validations / and translations

### PR DESCRIPTION
**Pull request by jagDhillon077**

### CODE REVIEW

File: packages/api/src/controllers/sensorController.js [hunk 1]
```diff
@@ -310,49 +312,60 @@ const sensorController = {
...
         depth,
-        grid_points: { lat: latitude, lng: longtitude },
         model,
-        part_number,
-        hardware_version,
       };
 
+      const figureID = await FigureModel.query()
+        .select('figure_id')
+        .where('location_id', location_id);
```
Comment (AbdusShaikh) [inline packages/api/src/controllers/sensorController.js:355]: We can use the baseControllers 'getByFieldId' function for this

File: packages/api/src/controllers/sensorController.js [hunk 2]
```diff
@@ -310,49 +312,60 @@ const sensorController = {
...
-      // const isSoilWaterContentActive = Object.entries(y[0])[1][1].active
-      // const isSoilWaterPotentialActive = Object.entries(y[1])[1][1].active
-      // const isTemperatureActive = Object.entries(y[2])[1][1].active
+
+      // Data is formatted in nested object values, these 5 const's are accessing reading types by using
+      // Object.entries and accessing each values via array indexing
+
+      if (reading_types.length !== 0) {
+        const x = Object.entries(reading_types);
+        const y = Object.entries(x[0][1]);
```
Comment (AbdusShaikh) [inline packages/api/src/controllers/sensorController.js:325]: We should have more descriptive variable names than x & y

File: packages/webapp/src/Routes.jsx [hunk 3]
```diff
@@ -553,8 +553,8 @@ const Routes = () => {
             <Route path="/fence/:location_id" component={FenceDetails} />
             <Route path="/buffer_zone/:location_id" component={BufferZoneDetails} />
             <Route path="/watercourse/:location_id" component={WatercourseDetails} />
-            <Route path="/sensor/:location_id" component={SensorDetails} />
-
+            <Route path="/sensor/:location_id/details" exact component={SensorDetail} />
+            <Route path="/sensor/:location_id/edit" exact component={EditSensor} />
```
Comment (AbdusShaikh) [inline packages/webapp/src/Routes.jsx:557]: The route should still be `<Route path="/sensor/:location_id" component={SensorDetails} />`. SensorDetails is a collection of routes that holds the rest of the sensor view routes. Similar to WatercourseDetails

File: packages/webapp/src/components/Sensor/EditSensor.jsx [hunk 4]
```diff
@@ -136,33 +144,25 @@ export default function UpdateSensor({
...
             hookFormRegister={register(LATITUDE, {
-              max: {
-                value: 90,
-                message: t('SENSOR.VALIDATION.SENSOR_LATITUDE'),
-              },
-              min: {
-                value: -90,
+              pattern: {
+                value:
+                  /^(\+|-)?(?:90(?:(?:\.0{1,10})?)|(?:[0-9]|[1-8][0-9])(?:(?:\.[0-9]{1,10})?))$/,
```
Comment (AbdusShaikh) [inline packages/webapp/src/components/Sensor/EditSensor.jsx:149]: A short comment about what this regex is doing would be helpful

File: packages/webapp/src/components/Sensor/EditSensor.jsx [hunk 5]
```diff
@@ -136,33 +144,25 @@ export default function UpdateSensor({
...
             hookFormRegister={register(LONGTITUDE, {
-              max: {
-                value: 180,
-                message: t('SENSOR.VALIDATION.SENSOR_LONGITUDE'),
-              },
-              min: {
-                value: -180,
+              pattern: {
+                value:
+                  /^(\+|-)?(?:180(?:(?:\.0{1,10})?)|(?:[0-9]|[1-9][0-9]|1[0-7][0-9])(?:(?:\.[0-9]{1,10})?))$/,
```
Comment (AbdusShaikh) [inline packages/webapp/src/components/Sensor/EditSensor.jsx:162]: A short comment about what this regex is doing would be helpful

**Review by AbdusShaikh — CHANGES_REQUESTED** [review]
I'm getting `ValidationError: point.lat: should be number, point.lng: should be number` when I try to update a sensor.
It says the error is coming from line 359 in `sensorController`

**Review by AbdusShaikh — CHANGES_REQUESTED** [review]
Looks good, though the locations table doesn't seem to be updated correctly. When I updated the name of one of my sensors it changed in the sensor table but not in the location table.

**Review by AbdusShaikh — CHANGES_REQUESTED** [review]
Looks good. Code is good and update is working properly

**Review by AbdusShaikh — APPROVED** [review]
Looks good

**Review by smattingly — APPROVED** [review]

**Pull request merged**